akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -233,6 +234,13 @@ abstract class GraphStageLogic private[stream] (val inCount: Int, val outCount:
 
   /**
    * INTERNAL API
+   *
+   * If possible a link back to the stage that the logic was created with, used for debugging.
+   */
+  private[stream] var originalStage: OptionVal[GraphStageWithMaterializedValue[_ <: Shape, _]] = OptionVal.None
```

Comment:
yes, I wanted to propose this. While this inflates the memory usage of the stage, we had array fields before that stored the stages anyway.